opcua/server/uaprocessor.py

```diff
@@ -95,8 +94,7 @@ def process(self, header, body):
 
         elif isinstance(msg, ua.ErrorMessage):
             self.logger.warning("Received an error message type")
-
-        else:
+        elif msg is not None:  # ua.MessageType.SecureMessage and ChunkType.Intermediate
             self.logger.warning("Unsupported message type: %s", header.MessageType)
             raise utils.ServiceError(ua.StatusCodes.BadTcpMessageTypeInvalid)
         return True
```

Comment:
This looks hard to follow.  We should have an "else" clause and explicitely test for the None case if it is something that happens. But if you really think that the current solution is better I trust you.